bundle/scripts/scripts.go

```diff
@@ -0,0 +1,68 @@
+package scripts
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/libs/cmdio"
+)
+
+func Execute(hook config.ScriptHook) bundle.Mutator {
+	return &script{
+		scriptHook: hook,
+	}
+}
+
+type script struct {
+	scriptHook config.ScriptHook
+}
+
+func (m *script) Name() string {
+	return fmt.Sprintf("scripts.%s", m.scriptHook)
+}
+
+func (m *script) Apply(ctx context.Context, b *bundle.Bundle) error {
+	out, err := executeHook(ctx, b, m.scriptHook)
+	cmdio.LogString(ctx, bytes.NewBuffer(out).String())
+	return err
+}
+
+func executeHook(ctx context.Context, b *bundle.Bundle, hook config.ScriptHook) ([]byte, error) {
+	command := getCommmand(b, hook)
+	if command == "" {
+		return nil, nil
+	}
+	commands := strings.Split(strings.ReplaceAll(string(command), "\r\n", "\n"), "\n")
+
+	out := make([][]byte, 0)
+	for _, command := range commands {
+		if command == "" {
+			continue
+		}
+
+		subcommands := strings.Split(command, " && ")
+		for _, subcommand := range subcommands {
+			buildParts := strings.Split(subcommand, " ")
```

Comment:
To make this approach work really well, please have a look at some real-life scripts in e.g. https://github.com/databricks/universe/blob/master/webapp/web/package.json:
* Scripts may use quotes in executables and parameters
* Scripts may use ';' or '|' (we should show a nice error if they do this)
* Scripts may do other weird things that we should probably just error on; customers can use a shell if they want weird things

Reply:
moved to using `sh` in this  commit https://github.com/databricks/cli/pull/632/commits/17d4472bd9515d59fd6670a8866b9a9e9a6db0e1 to avoid lexing altogether